src/ledger-iframe-bridge.ts

```diff
@@ -22,46 +22,57 @@ export enum IFrameMessageAction {
   LedgerSignTypedData = 'ledger-sign-typed-data',
 }
 
-type IFrameMessageResponse<TAction extends IFrameMessageAction> = {
-  action: TAction;
+type IFrameMessageResponseBase<
```

Comment:
Based on the name of this type, it seems like it would apply to all IFrameMessageResponses. But it only applies to some of them. Is another name we can use?

Or — maybe we just copy this pattern to all four types that use it? It seems like there are some slight variances among these types that we could consolidate down the road. Maybe we decide to make some kind of base type then.

What do you think?

Reply:
The advantage of extracting the base type imo is that it becomes instantly apparent which types conform to a common shape and which are atypical. I think maintaining that distinction provides clarity, especially if there's room for the atypical types to be adjusted down the road.

Would a different name like `IFrameMessageResponseStub` make things less confusing/misleading? If not, I'm also good with copy-pasting the pattern and maybe leaving a comment regarding consolidation.

Reply:
Fixed in https://github.com/MetaMask/eth-ledger-bridge-keyring/pull/207/commits/a42abc72338e63e566b2289fc8ea01d91afd737b